#VRML_SIM R2021a utf8
WorldInfo {
  ERP 0.8
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "wheel"
      material2 "pole"
      coulombFriction [
        -1
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5082577724460264 0.7496953110669407 0.42382894817655686 1.5538133056844288
  position 1.2896310957907677 2.8447950755217413 0.4193203341030124
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 3 3
}
Solid {
  translation 0 1 0
  children [
    DEF POLE Group {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.652857 0.64683 0.657359
            metalness 0
          }
          geometry Cylinder {
            radius 0.03
          }
        }
        Transform {
          translation 0 1 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.710491 0.703914 0.715419
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.5 0.001 0.5
              }
            }
          ]
        }
      ]
    }
  ]
  name "pole"
  contactMaterial "pole"
  boundingObject USE POLE
}
Robot {
  translation 0.00012361484533073702 0.301285721364593 0.00030935845777221426
  rotation -0.9859365042978618 -0.03579946557648022 0.16324095000137195 0.003205103010851192
  children [
    Solid {
      children [
        DEF BODY Group {
          children [
            DEF MOTOR_DISPLAY_1 Transform {
              translation -0.086827 0.111023 -0.0296244
              rotation -0.44706108459536525 0.44706208459555447 -0.7747721466066607 1.82325
              children [
                DEF MOTOR_MOCK Shape {
                  appearance PBRAppearance {
                    baseColor 1 0.7 0
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.008
                  }
                }
              ]
            }
            DEF MOTOR_DISPLAY_2 Transform {
              translation 0.067929 0.11 -0.0623199
              rotation -0.4470590447982244 -0.4470590447982244 0.7747750776375028 1.82325
              children [
                USE MOTOR_MOCK
              ]
            }
            DEF MOTOR_DISPLAY_3 Transform {
              translation 0.02 0.11 0.09
              rotation 0 0 1 -1.5707953071795862
              children [
                USE MOTOR_MOCK
              ]
            }
            DEF PI_BOARD_BAY Transform {
              translation -0.0565695 0.1 -0.101955
              rotation 0 1 0 1.0467
              children [
                DEF PI_BAY Shape {
                  appearance PBRAppearance {
                    baseColor 0.3 0.7 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.0856 0.011 0.026
                  }
                }
              ]
            }
            DEF MOTOR_BAY_1 Transform {
              translation 0.0529161 0.1 -0.0882933
              rotation 0 1 0 -1.0467
              children [
                DEF MOTOR_BAY Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.9
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.105 0.011 0.034
                  }
                }
              ]
            }
            DEF MOTOR_BAY_2 Transform {
              translation 0.05 0.1 0.09
              children [
                USE MOTOR_BAY
              ]
            }
            DEF MOTOR_BAY_3 Transform {
              translation -0.102959 0.1 -0.00171543
              rotation 0 1 0 1.0467
              children [
                USE MOTOR_BAY
              ]
            }
            DEF SMALL_JOINT_3 Transform {
              translation -0.12985 0.1 0.0750989
              rotation 0 1 0 -1.0467
              children [
                DEF SMALL_BODY_SIDE Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.03 0.01
                  }
                }
              ]
            }
            DEF SMALL_JOINT_2 Transform {
              translation 0.12985 0.1 0.075099
              rotation 0 1 0 1.0467
              children [
                USE SMALL_BODY_SIDE
              ]
            }
            DEF BACK_PLATE_1 Transform {
              translation 0 0.1 0.11
              children [
                DEF BODY_SIDE Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.221 0.03 0.01
                  }
                }
              ]
            }
            DEF SMALL_JOINT_1 Transform {
              translation 0 0.1 -0.15
              children [
                USE SMALL_BODY_SIDE
              ]
            }
            DEF BACK_PLATE_2 Transform {
              translation -0.0952532 0.1 -0.0550129
              rotation 0 1 0 1.0467
              children [
                DEF TEST Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.221 0.03 0.01
                  }
                }
              ]
            }
            DEF BACK_PLATE_3 Transform {
              translation 0.0952532 0.1 -0.0550129
              rotation 0 1 0 -1.0467
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.221 0.03 0.01
                  }
                }
              ]
            }
            Transform {
              translation 0.0175 0.1 0.055
              children [
                DEF BODY_CONNECTOR_SHORT Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.01 0.04
                  }
                }
              ]
            }
            Transform {
              translation -0.0175 0.1 0.075
              children [
                DEF BODY_CONNECTOR_LONG Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.01 0.079
                  }
                }
              ]
            }
            Transform {
              translation -0.0562046 0.1 -0.052653
              rotation 0 1 0 1.0476
              children [
                USE BODY_CONNECTOR_LONG
              ]
            }
            Transform {
              translation -0.0564 0.1 -0.0123
              rotation 0 1 0 1.0476
              children [
                USE BODY_CONNECTOR_SHORT
              ]
            }
            Transform {
              translation 0.03888 0.1 -0.04266
              rotation 0 1 0 -1.0467
              children [
                USE BODY_CONNECTOR_SHORT
              ]
            }
            Transform {
              translation 0.0737156 0.1 -0.0223086
              rotation 0 1 0 -1.0467
              children [
                USE BODY_CONNECTOR_LONG
              ]
            }
            Transform {
              translation 0 0.1 0.045
              rotation 0 0 1 1.5708
              children [
                DEF AXIS Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.3
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.039
                    radius 0.003
                  }
                }
              ]
            }
            Transform {
              translation -0.03897 0.1 -0.0225
              rotation -0.866019052628739 0 -0.5000110003630134 1.5707999999999998
              children [
                USE AXIS
              ]
            }
            Transform {
              translation 0.03897 0.1 -0.0225
              rotation 0.8660190526287391 0 -0.5000110003630134 1.5708
              children [
                USE AXIS
              ]
            }
          ]
        }
        Group {
          children [
            SolidPipe {
              translation 0 0.13 0
              name "UV module"
              height 0.02
              radius 0.05
              thickness 0.005
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              physics Physics {
              }
            }
            Solid {
              translation 0.088 0.12 0.05
              rotation 0.7255771900217981 -0.5459391429762939 -0.4189131097093782 -1.6267153071795866
              children [
                DEF UV_CONNECTOR Shape {
                  appearance PBRAppearance {
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.11
                    radius 0.004
                  }
                }
              ]
            }
            Transform {
              translation 0 0.12 -0.099
              rotation 1 0 0 1.309
              children [
                USE UV_CONNECTOR
              ]
            }
            Transform {
              translation -0.087 0.12 0.05
              rotation 0.7255801029453123 0.5459360774574161 0.41891205943524723 -1.626705307179586
              children [
                USE UV_CONNECTOR
              ]
            }
          ]
        }
      ]
      boundingObject USE BODY
      physics Physics {
      }
    }
    LED {
      translation 0 0.13 0
      children [
        SolidPipe {
          name "leds"
          height 0.01
          radius 0.045
          thickness 0.003
          appearance PBRAppearance {
            baseColor 0.7 0.7 0.7
            roughness 0.3
            metalness 0
            emissiveIntensity 6
          }
        }
      ]
      name "uv_led"
      color [
        0.5 0 1
      ]
      gradual TRUE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 12.799947167332826
        axis -1 0 0
        anchor 0 0.1 0.045
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation -3.492307801909842e-08 0.09999785484740256 0.045000698439504966
        rotation 0.055975110718735184 -0.05597534452758023 -0.9968618498994954 4.709242747749159
        children [
          DEF WHEEL_CENTRE Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Cylinder {
              height 0.021
              radius 0.012
            }
          }
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.016
            }
          }
        ]
        name "wheel_shape1"
        contactMaterial "wheel"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
        linearVelocity 4.119057680348081e-05 -0.07823070420161152 0.00023305584740769558
        angularVelocity 4.997833524972641 0.002937662104500114 -0.0034445366258787132
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 12.79992103277773
        axis 0.5 0 0.866
        anchor 0.039 0.1 -0.0225
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.039000870456695334 0.09999839911106488 -0.022500588813684948
        rotation -0.8912839399688982 -0.05596491948794216 0.44997873965358115 4.709242233445011
        children [
          USE WHEEL_CENTRE
          USE WHEEL
        ]
        name "wheel_shape2"
        contactMaterial "wheel"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
        linearVelocity 3.606314165110555e-05 -0.07853244965462025 0.00024868238375778604
        angularVelocity -2.501730828298849 -0.015057971155727063 -4.334307930775776
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 12.799895922799493
        axis 0.5 0 -0.866
        anchor -0.039 0.1 -0.0225
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.03900118272702512 0.0999984428527138 -0.02250074149773075
        rotation 0.8353272212649392 -0.05594892912792167 0.5468986658817466 4.70924589649938
        children [
          USE WHEEL_CENTRE
          USE WHEEL
        ]
        name "wheel_shape3"
        contactMaterial "wheel"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
        linearVelocity 4.6444856542076e-05 -0.07822561725920356 0.0002471799874479214
        angularVelocity -2.502780943342917 0.01206893180765299 4.325800557126215
      }
    }
    DistanceSensor {
      translation 0 0.115 0.11
      rotation 0 0 1 1.5708
      children [
        DEF D_SENSOR Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.01 0.005
          }
        }
      ]
      name "ds_top"
      boundingObject USE D_SENSOR
      physics Physics {
      }
      numberOfRays 4
      aperture 2
      gaussianWidth 7
    }
    DistanceSensor {
      translation 0 0.085 0.105
      rotation 0 0 1 -1.5708
      children [
        USE D_SENSOR
      ]
      name "ds_bottom"
      boundingObject USE D_SENSOR
      physics Physics {
      }
      numberOfRays 4
      aperture 2
      gaussianWidth 7
    }
  ]
  name "PAUL"
  boundingObject USE BODY
  physics Physics {
  }
  rotationStep 0.1
  controller "PAUL_withLED"
  linearVelocity 4.598171022863516e-05 -0.07832257788312375 0.00024152603474916778
  angularVelocity -0.0021659320887111085 -1.6949488118958247e-05 -0.0039552558561120185
}
